### PR TITLE
Add ORCA GUC optimizer_enable_nljoin to disable NLJ

### PR DESCRIPTION
**Pull request by dgkimura**
Use optimizer_enable_nljoin to disable all xforms that produce nestloop
join alternatives.

Co-authored-by: Orhan Kislal <okislal@vmware.com>

### CODE REVIEW

File: src/backend/utils/misc/guc_gp.c [hunk 1]
```diff
@@ -2002,6 +2003,16 @@ struct config_bool ConfigureNamesBool_gp[] =
 		true,
 		NULL, NULL, NULL
 	},
+	{
+		{"optimizer_enable_nljoin", PGC_USERSET, DEVELOPER_OPTIONS,
+			gettext_noop("Enable nested loops join plans in the optimizer."),
+			NULL,
+			GUC_NO_SHOW_ALL | GUC_NOT_IN_SAMPLE
```
Comment (bhuvnesh2703) [inline src/backend/utils/misc/guc_gp.c:2010]: if you add these flags, and run gpconfig -c optimizer_enable_nljoin -v on, it potentially will fail and ask to pass --skipvalidation flag, but ideally we shouldn't. So could you please see that we don't have to pass --skipvalidation option.

File: src/backend/utils/misc/guc_gp.c [hunk 2]
```diff
@@ -2002,6 +2003,16 @@ struct config_bool ConfigureNamesBool_gp[] =
 		true,
 		NULL, NULL, NULL
 	},
+	{
+		{"optimizer_enable_nljoin", PGC_USERSET, DEVELOPER_OPTIONS,
```
Comment (bhuvnesh2703) [inline src/backend/utils/misc/guc_gp.c:2007]: Instead of `DEVELOPER_OPTIONS`,  should it be `QUERY_TUNING_METHOD`?

Comment (dgkimura) [reply src/backend/utils/misc/guc_gp.c]: Yeah.  Added commit to update all ORCA GUCs that enable/disable physical operators to be `QUERY_TUNING_METHOD`.

File: src/backend/utils/misc/guc_gp.c (hunk 1 above)
Comment (dgkimura) [reply src/backend/utils/misc/guc_gp.c]: Yeah, need to remove `GUC_NO_SHOW_ALL` flag to make that work.  Added commit that does the same for all other ORCA GUCs that enable/disable physical operator

**Review by bhuvnesh2703 — APPROVED** [review]

**Pull request merged**